Framework/PythonInterface/mantid/plots/mantidaxes.py

```diff
@@ -450,8 +450,12 @@ def replot_artist(self, artist, errorbars=False, **kwargs):
         kwargs['distribution'] = not self.get_artist_normalization_state(artist)
         workspace, spec_num = self.get_artists_workspace_and_spec_num(artist)
 
+        # deal with MDHisto workspace
+        if workspace.isMDHistoWorkspace():
+            if 'distribution' in kwargs.keys():
+                del kwargs['distribution']
         # check if it is a sample log plot
-        if spec_num is None:
+        elif spec_num is None:
             sample_log_plot_details = self.get_artists_sample_log_plot_details(artist)
             kwargs['LogName'] = sample_log_plot_details[0]
```

Comment:
_sample_log_plot_details_ can be **None**.  There is no check for this.  And so far, the logic in the code does not prevent this value to be **None**.